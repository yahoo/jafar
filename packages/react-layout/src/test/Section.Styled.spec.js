/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import renderer from 'react-test-renderer';
import Style from '../components/Section/Styled';
import 'jest-styled-components';

describe('Styled', () => {
  [
    { size: 1, level: 1 },
    { size: 1, level: 2 },
    { size: 2, level: 1 },
    { size: 2, level: 2 },
    { size: 3, level: 1 },
    { size: 3, level: 2 },
    { size: 4, level: 1 },
    { size: 4, level: 2 },
  ]
    .forEach(({ size, level }) => {
      describe(`size: ${size}, level: ${level}`, () => {
        const Styled = Style(size, level);
  
        it('Wrapper', () => {
          const tree = renderer.create(<Styled.Wrapper />).toJSON();
          expect(tree).toMatchSnapshot();
        });
  
        it('Title', () => {
          const tree = renderer.create(<Styled.Title />).toJSON();
          expect(tree).toMatchSnapshot();
        });
      });
    });  
});
