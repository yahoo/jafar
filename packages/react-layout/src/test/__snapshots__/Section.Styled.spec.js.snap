// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled size: 1, root: false Title 1`] = `
.c0 {
  text-transform: uppercase;
  line-height: 1.1;
  color: inherit;
  font-size: 16px;
  font-weight: 400;
  padding-bottom: 24px;
  margin: 0;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 1, root: false Wrapper 1`] = `
.c0 {
  padding: 0;
  border-bottom: none;
}

.c0:last-child {
  border-bottom: none;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 1, root: true Title 1`] = `
.c0 {
  text-transform: uppercase;
  line-height: 1.1;
  color: inherit;
  font-size: 16px;
  font-weight: 500;
  padding-bottom: 24px;
  margin: 0;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 1, root: true Wrapper 1`] = `
.c0 {
  padding: 26px 20px 10px 20px;
  border-bottom: 1px solid #d8d8d8;
}

.c0:last-child {
  border-bottom: none;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 2, root: false Title 1`] = `
.c0 {
  text-transform: uppercase;
  line-height: 1.1;
  color: inherit;
  font-size: 18px;
  font-weight: 400;
  padding-bottom: 30px;
  margin: 0;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 2, root: false Wrapper 1`] = `
.c0 {
  padding: 0;
  border-bottom: none;
}

.c0:last-child {
  border-bottom: none;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 2, root: true Title 1`] = `
.c0 {
  text-transform: uppercase;
  line-height: 1.1;
  color: inherit;
  font-size: 18px;
  font-weight: 500;
  padding-bottom: 30px;
  margin: 0;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 2, root: true Wrapper 1`] = `
.c0 {
  padding: 34px 30px 20px 30px;
  border-bottom: 1px solid #d8d8d8;
}

.c0:last-child {
  border-bottom: none;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 3, root: false Title 1`] = `
.c0 {
  text-transform: uppercase;
  line-height: 1.1;
  color: inherit;
  font-size: 20px;
  font-weight: 400;
  padding-bottom: 40px;
  margin: 0;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 3, root: false Wrapper 1`] = `
.c0 {
  padding: 0;
  border-bottom: none;
}

.c0:last-child {
  border-bottom: none;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 3, root: true Title 1`] = `
.c0 {
  text-transform: uppercase;
  line-height: 1.1;
  color: inherit;
  font-size: 20px;
  font-weight: 500;
  padding-bottom: 40px;
  margin: 0;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 3, root: true Wrapper 1`] = `
.c0 {
  padding: 42px 40px 30px 40px;
  border-bottom: 1px solid #d8d8d8;
}

.c0:last-child {
  border-bottom: none;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 4, root: false Title 1`] = `
.c0 {
  text-transform: uppercase;
  line-height: 1.1;
  color: inherit;
  font-size: 22px;
  font-weight: 400;
  padding-bottom: 50px;
  margin: 0;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 4, root: false Wrapper 1`] = `
.c0 {
  padding: 0;
  border-bottom: none;
}

.c0:last-child {
  border-bottom: none;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 4, root: true Title 1`] = `
.c0 {
  text-transform: uppercase;
  line-height: 1.1;
  color: inherit;
  font-size: 22px;
  font-weight: 500;
  padding-bottom: 50px;
  margin: 0;
}

<div
  className="c0"
/>
`;

exports[`Styled size: 4, root: true Wrapper 1`] = `
.c0 {
  padding: 50px 50px 40px 50px;
  border-bottom: 1px solid #d8d8d8;
}

.c0:last-child {
  border-bottom: none;
}

<div
  className="c0"
/>
`;
